sidechain/consensus/slots/src/lib.rs

```diff
@@ -203,25 +222,67 @@ pub trait SimpleSlotWorker<ParentchainBlock: ParentchainBlockTrait> {
 			return None
 		}
 
-		let latest_parentchain_header = match self.peek_latest_parentchain_header() {
-			Ok(Some(peeked_header)) => peeked_header,
-			Ok(None) => slot_info.last_imported_parentchain_head.clone(),
-			Err(e) => {
-				warn!(
-					target: logging_target,
-					"Failed to peek latest parentchain block header: {:?}", e
-				);
-				return None
-			},
-		};
+		let latest_integritee_parentchain_header =
+			match self.peek_latest_integritee_parentchain_header() {
+				Ok(Some(peeked_header)) => peeked_header,
+				Ok(None) => slot_info.last_imported_integritee_parentchain_head.clone(),
+				Err(e) => {
+					warn!(
+						target: logging_target,
+						"Failed to peek latest Integritee parentchain block header: {:?}", e
+					);
+					return None
+				},
+			};
+		trace!(
+			target: logging_target,
+			"on_slot: a priori latest Integritee block number: {:?}",
+			latest_integritee_parentchain_header.number()
+		);
+		// fixme: we need proper error handling here. we just assume there is no target_a if there is an error here, which is very brittle
+		let maybe_latest_target_a_parentchain_header =
```

Comment:
FYI

Reply:
We might want to pass a global runtime config at startup into the enclave, that can be referenced later to know some runtime args.

Reply:
#1511